front/lib/resources/mcp_server_connection_resource.ts

```diff
@@ -208,12 +209,15 @@ export class MCPServerConnectionResource extends BaseResource<MCPServerConnectio
 
     // Only return the latest connection for a given MCP server.
     // Ideally we would filter in the query directly.
-    const connectionsByServerId = keyBy(
```

Comment:
keyBy takes the last encountered item for a key, reversing a list that's sorted newest-first will make the newest item the last item for each server ID group -> meaning on this function we had a valid desc order but very sneaky 😅